pink/rondis/string/db_operations.cc

```diff
@@ -614,3 +614,164 @@ int rondb_get_rondb_key(const NdbDictionary::Table *tab,
+    insert_value_ptr[0] = 1; // Length is 1
+    insert_value_ptr[1] = 0; // Second length byte is 0
+    insert_value_ptr[2] = '1'; //Inserts a string '1'
+    insert_value_ptr[3] = 0;
+
+    code.load_const_mem(REG0, REG2, 3, &insert_value);// Load to memory
+    code.write_from_mem(value_start_col, REG6, REG2); // Write to column
+    code.write_attr(tot_value_len_col, REG3);
+    code.interpret_exit_ok();
+
+    /* Program end, now compile code */
+    ret_code = code.finalise();
+    if (ret_code != 0) {
+        assign_ndb_err_to_response(response,
+                                   "Failed to create Interpreted code",
+                                   code.getNdbError());
+        return;
+    }
+
+    /* Prepare the interpreted program to be part of the write */
+    opts.optionsPresent |= NdbOperation::OperationOptions::OO_INTERPRETED;
+    opts.optionsPresent |= NdbOperation::OperationOptions::OO_INTERPRETED_INSERT;
+    opts.interpretedCode = &code;
+
+    /**
+     * Prepare to get the final value of the Redis row after INCR is finished
+     * This is performed by the reading the pseudo column that is reading the
+     * output index written in interpreter program.
+     */
+    NdbOperation::GetValueSpec getvals[1];
+    getvals[0].appStorage = nullptr;
+    getvals[0].recAttr = nullptr;
+    getvals[0].column = NdbDictionary::Column::READ_INTERPRETER_OUTPUT_0;
+    opts.optionsPresent |= NdbOperation::OperationOptions::OO_GET_FINAL_VALUE;
+    opts.numExtraGetFinalValues = 1;
+    opts.extraGetFinalValues = getvals;
+
+    /* Define the actual operation to be sent to RonDB data node. */
+    const NdbOperation *op = trans->writeTuple(
+      pk_key_record,
+      (const char*)key_row,
+      entire_key_record,
+      (char*)key_row,
+      mask_ptr,
+      &opts,
+      sizeof(opts));
+    if (op == nullptr) {
+        assign_ndb_err_to_response(response,
+                                   "Failed to create NdbOperation",
+                                   trans->getNdbError());
+        return;
+    }
+
+    /* Send to RonDB and execute the INCR operation */
+    if (trans->execute(NdbTransaction::Commit,
+                       NdbOperation::AbortOnError) != 0 ||
+        trans->getNdbError().code != 0)
+    {
+        assign_ndb_err_to_response(response,
+                                   FAILED_INCR_KEY,
+                                   trans->getNdbError());
+        return;
+    }
+
+    /* Retrieve the returned new value as an Int64 value */
+    NdbRecAttr *recAttr = getvals[0].recAttr;
+    Int64 new_incremented_value = recAttr->int64_value();
+
+    /* Send the return message to Redis client */
+    char header_buf[20];
+    int header_len = write_formatted(header_buf,
+                                     sizeof(header_buf),
+                                     ":%lld\r\n",
+                                     new_incremented_value);
+    response->assign(header_buf);
+    return;
```

Comment:
What if the row is not an integer? Can you check that in the interpreted program and exit early with a corresponding error message?

Reply:
If the value_start isn't a string representing an integer I will discover that in str_to_int64 execution in the RonDB data node and report back an error code when executing the write operation.